### PR TITLE
feat: agent-chart

### PR DESCRIPTION
**Pull request by yashmehrotra**
Fixes: #112

### CODE REVIEW

File: agent-chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Use this only if you want to replace the default that is .Chart.Name as the name of all the objects.
+nameOverride: ""
+
+# If true, creates secret
+upstream:
+  createSecret: false
```
Comment (yashmehrotra) [inline agent-chart/values.yaml:6]: @moshloop Should the default here be true ?

We do check all the necessary values in templates/required.yaml

File: agent-chart/values.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Use this only if you want to replace the default that is .Chart.Name as the name of all the objects.
+nameOverride: ""
```
Comment (moshloop) [inline agent-chart/values.yaml:2]:
```suggestion
nameOverride: "mc-agent"
```

**Comment by moshloop** [comment]
@yashmehrotra do we not need a workflow to release ?

File: agent-chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  image:
+    type: full
+  upstream:
+    enabled: true
+    agentName: "default-agent"
+    secretKeyRef:
+      name: upstream
+  db:
+    embedded:
+      persist: true
```
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply agent-chart/values.yaml:22]: This needs to be true, else a stateful set won't be created

**Comment by yashmehrotra** [comment]
> @yashmehrotra do we not need a workflow to release ?
@moshloop Yeah, will do that in a separate PR along with updating canary and config-db's workflow to update both the charts

File: agent-chart/values.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply agent-chart/values.yaml:22]: If persist is false, embedded postgres will run in Deployment and we can lose data if pod restarts/crashes or even reschedules

**Pull request merged**